src/openforms/appointments/views.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist, PermissionDenied
+from django.utils.http import urlsafe_base64_decode
+from django.utils.translation import gettext_lazy as _
+from django.views.generic import RedirectView
+
+from drf_spectacular.utils import OpenApiResponse, extend_schema
+
+from openforms.api.serializers import ExceptionSerializer
+from openforms.config.models import GlobalConfiguration
+from openforms.submissions.models import Submission
+from openforms.submissions.utils import add_submmission_to_session
+
+from .tokens import submission_appointment_token_generator
+
+logger = logging.getLogger(__name__)
+
+
+@extend_schema(
+    summary=_("Verify the appointment cancel link."),
+    responses={
+        302: None,
+        403: OpenApiResponse(
+            response=ExceptionSerializer,
+            description=_("Unable to verify token."),
+        ),
+    },
+)
+class VerifyCancelAppointmentLinkView(RedirectView):
+
+    permission_classes = ()
+    authentication_classes = ()
+
+    def get_redirect_url(self, submission_uuid: int, token: str, *args, **kwargs):
+        try:
+            submission = Submission.objects.get(uuid=submission_uuid)
+        except ObjectDoesNotExist:
+            logger.debug(
+                "Called endpoint with an invalid submission uuid: %s", submission_uuid
+            )
+            raise PermissionDenied("Cancel url is not valid")
+
+        # Check that the token is valid
+        valid = submission_appointment_token_generator.check_token(submission, token)
+        if not valid:
+            logger.debug("Called endpoint with an invalid token: %s", token)
+            raise PermissionDenied("Cancel url is not valid")
+
+        add_submmission_to_session(submission, self.request.session)
+
+        config = GlobalConfiguration.get_solo()
+
+        return (
+            f"{config.cancel_appointment_page}"
+            f"?time={submission.appointment_info.start_time.isoformat()}"
+            f"&submission_uuid={str(submission_uuid)}"
+        )
```

Comment:
This is not how you build URLs safely and there is plenty of tooling at your hands to do it the right way:

1. `furl`, mentioned earlier
2. python's stdlib `urlib` package
3. Django's own utils: https://docs.djangoproject.com/en/2.2/ref/utils/#django.utils.http.urlencode

Also - don't go crazy with f-strings, the fact that you can evaluate about anything between the curly braces doesn't mean you should. It should only really be used for drop-in variables, so bring those variables/method calls outside of the f-string.

besides that, doing `f"{str(uuid)}"` is literally the same effect as doing the more readable `f"{uuid}"`, as the string representation will be used by the f-string feature.